plasma_framework/test/src/exits/payment/PaymentChallengeIFEInputSpent.test.js

```diff
@@ -0,0 +1,357 @@
+        const challengeIFEInputSpent = await PaymentChallengeIFEInputSpent.new();
+
+        await PaymentInFlightExitRouter.link('PaymentStartInFlightExit', startInFlightExit.address);
+        await PaymentInFlightExitRouter.link('PaymentPiggybackInFlightExit', piggybackInFlightExit.address);
+        await PaymentInFlightExitRouter.link('PaymentChallengeIFENotCanonical', challengeInFlightExitNotCanonical.address);
+        await PaymentInFlightExitRouter.link('PaymentChallengeIFEInputSpent', challengeIFEInputSpent.address);
+
+        this.exitIdHelper = await ExitId.new();
+        this.stateTransitionVerifierAccept = await StateTransitionVerifierAccept.new();
+    });
+
+    describe('challenge in-flight exit input spent', () => {
+        // This is the transaction whose output is the input piggyback in the IFE.
+        function buildInputTx() {
+            const tx = createInputTransaction(
+                [buildUtxoPos(2000, 4, 3), buildUtxoPos(2000, 10, 2)],
+                inputOwner,
+                334455,
+            );
+
+            const txBytes = web3.utils.bytesToHex(tx.rlpEncoded());
+            const outputIndex = 0;
+
+            return {
+                tx,
+                txBytes,
+                outputIndex,
+                utxoPos: buildUtxoPos(3000, 5, outputIndex),
+            };
+        }
+
+        // Sets up an IFE tx using inputTx as an input, starts the IFE and piggybacks on the input.
+        const buildPiggybackInputData = async (inputTx) => {
+            const outputAmount = 997;
+
+            const firstInput = createInputTransaction([buildUtxoPos(BLOCK_NUMBER, 3, 0)], outputOwner, 334455);
+            const firstInputUtxoPos = buildUtxoPos(3000, 66, 0);
+
+            const inFlightTx = createInFlightTx(
+                [firstInput, inputTx.tx],
+                [firstInputUtxoPos, inputTx.utxoPos],
+                alice,
+                outputAmount,
+            );
+            const rlpInFlightTxBytes = web3.utils.bytesToHex(inFlightTx.rlpEncoded());
+
+            const emptyWithdrawData = {
+                outputId: web3.utils.sha3('dummy output id'),
+                outputGuard: web3.utils.sha3('dummy output guard'),
+                exitTarget: constants.ZERO_ADDRESS,
+                token: constants.ZERO_ADDRESS,
+                amount: 0,
+            };
+
+            const inFlightExitData = {
+                exitStartTimestamp: (await time.latest()).toNumber(),
+                exitMap: 0,
+                position: INFLIGHT_EXIT_YOUNGEST_INPUT_POSITION,
+                bondOwner: alice,
+                oldestCompetitorPosition: 0,
+                inputs: [{
+                    outputId: web3.utils.sha3('dummy output id'),
+                    outputGuard: web3.utils.sha3('dummy output guard'),
+                    exitTarget: inputOwner,
+                    token: ETH,
+                    amount: 999,
```

Comment:
Minor: The amount here does not match your input tx amount though. Same on another input amount.

Reply:
Well this one doesn't have to match because it's an unrelated random input.
The other one should probably match though...

Reply:
okay, I just have the tendency to make the test data as real as possible unless it's intended to test edge case so that whoever reads the tests won't be confused. But anyway, I am fine with that and it's not a big issue as long as test works.